ppsci/optimizer/soap.py

```diff
@@ -0,0 +1,554 @@
+        merge_dims (bool, optional):
+            Whether or not to merge dimensions of the preconditioner. defaults to `False`.
+        precondition_1d (bool, optional):
+            Whether or not to precondition 1D gradients. defaults to `False`.
+        normalize_grads (bool, optional):
+            Whether or not to normalize gradients per layer.
+            Helps at large precondition_frequency (~100 in our experiments),
+            but hurts performance at small precondition_frequency (~10 in our experiments). defaults to `False`.
+        data_format (str, optional):
+            Data format of the input for convolutional layers.
+            Should be "channels_last" for data_format of NHWC and "channels_first" for NCHW. defaults to `channels_first`.
+        correct_bias (bool, optional):
+            Whether or not to use bias correction in Adam. defaults to `True`.
+        name (str, optional): Normally there is no need for user to set this property.
+            For more information, please refer to :ref:`api_guide_Name`.
+            The default value is None.
+
+    Return:
+        loss (Tensor): the final loss of closure.
+
+    Examples:
+        .. code-block:: python
+
+            >>> import paddle
+            >>> import ppsci
+            >>> import numpy as np
+
+            >>> paddle.disable_static()
+            >>> np.random.seed(0)
+            >>> np_w = np.random.rand(1).astype(np.float32)
+            >>> np_x = np.random.rand(1).astype(np.float32)
+
+            >>> inputs = [np.random.rand(1).astype(np.float32) for i in range(10)]
+            >>> # y = 2x
+            >>> targets = [2 * x for x in inputs]
+
+            >>> class Net(paddle.nn.Layer):
+            ...     def __init__(self):
+            ...         super().__init__()
+            ...         w = paddle.to_tensor(np_w)
+            ...         self.w = paddle.create_parameter(shape=w.shape, dtype=w.dtype, default_initializer=paddle.nn.initializer.Assign(w))
+            ...
+            ...     def forward(self, x):
+            ...         return self.w * x
+            ...
+            >>> net = Net()
+            >>> opt = ppsci.optimizer.soap.SOAP(params=net.parameters())
+            >>> def train_step(inputs, targets):
+            ...     def closure():
+            ...         outputs = net(inputs)
+            ...         loss = paddle.nn.functional.mse_loss(outputs, targets)
+            ...         print('loss: ', loss.item())
+            ...         opt.clear_grad()
+            ...         loss.backward()
+            ...         return loss
+            ...     opt.step(closure)
+            ...
+            >>> for input, target in zip(inputs, targets):
+            ...     input = paddle.to_tensor(input)
+            ...     target = paddle.to_tensor(target)
+            ...     train_step(input, target)
+    """
+
+    def __init__(
+        self,
+        params,
+        lr: float = 3e-3,
```

Comment:
学习率的名字需要跟框架现有优化器保持一致
```suggestion
        learning_rate: float = 3e-3,
```

Reply:
ACK. Thx.